dashboards-observability/public/components/explorer/home_table/history_table.tsx

```diff
@@ -40,43 +39,100 @@ export function Table(options: TableData) {
   };
 
   const columns = [
+    {
+      field: 'type',
+      name: '',
+      width: '3%',
+      render: (item) => {
+        if (item == 'Visualization') {
+          return (
+            <div>
+              <EuiLoadingChart size="m" />
+            </div>
+          );
+        } else {
+          return (
+            <div>
+              <EuiIcon type="discoverApp" size="m" />
+            </div>
+          );
+        }
+      },
+    },
     {
       field: 'data',
       name: 'Name',
-      render: (item)=>{return <EuiLink onClick={() =>
-      {options.savedQuerySearch(item.query, [item.date_start, item.date_end], item.timestamp, item.fields)}}>
-        {item.name}
-      </EuiLink>},
+      align: 'left',
+      render: (item) => {
+        return (
+          <EuiLink
+            onClick={() => {
+              options.savedQuerySearch(
+                item.query,
+                [item.date_start, item.date_end],
+                item.timestamp,
+                item.fields
+              );
+            }}
+          >
+            {item.name}
+          </EuiLink>
+        );
+      },
     },
     {
-      field: 'description',
-      name: 'Description',
+      field: 'type',
+      name: 'Type',
+      align: 'right',
     },
   ];
 
-
-
+  let objectName = '';
   const queries = options.savedHistory.map((h) => {
     const savedObject = h.hasOwnProperty('savedVisualization')
       ? h.savedVisualization
       : h.savedQuery;
+    const object = h.hasOwnProperty('savedVisualization');
+    if (object) {
+      objectName = 'Visualization';
+    } else {
+      objectName = 'Query';
+    }
     return {
       data: {
         name: savedObject.name,
         query: savedObject.query,
         date_start: savedObject.selected_date_range.start,
-        date_end : savedObject.selected_date_range.end,
+        date_end: savedObject.selected_date_range.end,
         timestamp: savedObject.selected_timestamp?.name || '',
-        fields: savedObject.selected_fields?.tokens || []
+        fields: savedObject.selected_fields?.tokens || [],
       },
       name: savedObject.name || '',
-      description: savedObject.description || '',
-
+      type: objectName,
     };
   });
 
- 
   const totalItemCount = queries.length;
+  const filterOptions = ['Visualization', 'Query'];
```

Comment:
Maybe a constant like FILTER_OPTIONS = ['Visualization', 'Query'], and move it to constants/explorer.ts file?

Reply:
Yeah, i'll make that change